src/components/IconBadge/IconBadge.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import classNames from 'classnames'
+import { Props } from './IconBadge.types'
+import './IconBadge.css'
+
+const IconBadge = ({ icon, text, onClick, className, children }: Props) => {
+  const childrenInt = React.useMemo(
+    () => (
+      <>
+        {children ? (
+          <span className="custom-icon">{children}</span>
+        ) : (
+          <span className={classNames(icon && 'icon', icon)} />
+        )}
+        <span className="text">{text}</span>
+      </>
+    ),
+    [children, icon, text]
+  )
+
+  return (
+    <div className={classNames('IconBadge', className)} onClick={onClick}>
```

Comment:
As we're using the [BEM standard](https://getbem.com/naming/) for the UI classes now, this class should be:
```suggestion
    <div className={classNames('dui-icon-badge', className)} onClick={onClick}>
```

Reply:
done!